src/Raven.Server/Web/Studio/Sharding/Processors/ShardedStudioCollectionsHandlerProcessorForPreviewCollection.cs

```diff
@@ -84,12 +84,41 @@ protected override async ValueTask<long> GetTotalResultsAsync()
         return total;
     }
 
-    protected override bool NotModified(out string etag)
+    protected override async ValueTask<bool> NotModifiedAsync()
     {
-        DevelopmentHelper.ShardingToDo(DevelopmentHelper.TeamMember.Karmel, DevelopmentHelper.Severity.Normal,
-            "RavenDB-19066 Need to figure out the best way to combine ETags and send not modified. See `null` passed as etag to above new ShardedCollectionPreviewOperation()");
+        var changeVectors = new List<string>();
+        if (IsAllDocsCollection)
+        {
+            for (int i = 0; i < RequestHandler.DatabaseContext.ShardCount; i++)
+            {
+                var shardExecutor = _requestHandler.ShardExecutor.GetRequestExecutorAt(i);
+                var cmd = new GetStatisticsOperation.GetStatisticsCommand(RequestHandler.Server.DebugTag, RequestHandler.DatabaseContext.ServerStore.NodeTag);
+                await shardExecutor.ExecuteAsync(cmd, _context);
+                var changeVector = cmd.Result.DatabaseChangeVector;
+                if (changeVector != null)
+                    changeVectors.Add(changeVector);
+            }
+        }
+        else
+        {
+            var result = await _requestHandler.ShardExecutor.ExecuteParallelForAllAsync(new ShardedLastChangeVectorForCollectionOperation(RequestHandler, Collection, RequestHandler.DatabaseName));
+            foreach ((_, string changeVector) in result.LastChangeVectors)
+            {
+                if (changeVector != null)
+                    changeVectors.Add(changeVector);
+            }
+        }
```

Comment:
why not simply to `ShardedLastChangeVectorForCollectionOperation` for each shard?

Reply:
I don't think you need to do any requests `NotModified` ... you should be able to get it from the `ShardedCollectionPreviewOperation` request to each of the shard

Reply:
True. Done :)